glue/viewers/common/viewer.py

```diff
@@ -262,13 +270,15 @@ def _update_data(self, message):
                 else:
                     if layer_artist.layer is message.data:
                         layer_artist.update()
+        self.draw_legend()
```

Comment:
Is this actually needed since you have added a callback to ``layers``?

Reply:
Indeed, it can be removed from both _update_data and _update_subsets ! I'm not sure why I thought I needed them.